security/pkg/stsservice/tokenmanager/tokenmanager.go

```diff
@@ -80,22 +80,24 @@ func GetGCPProjectInfo() GCPProjectInfo {
 
 // CreateTokenManager creates a token manager with specified type and returns
 // that token manager
-func CreateTokenManager(tokenManagerType string, config Config) security.TokenManager {
+func CreateTokenManager(tokenManagerType string, config Config) (security.TokenManager, error) {
 	tm := &TokenManager{
 		plugin: nil,
 	}
+	var err error
 	switch tokenManagerType {
 	case GoogleTokenExchange:
 		if projectInfo := GetGCPProjectInfo(); len(projectInfo.Number) > 0 {
 			if p, err := google.CreateTokenManagerPlugin(config.CredFetcher, config.TrustDomain,
 				projectInfo.Number, projectInfo.clusterURL, true); err == nil {
 				tm.plugin = p
 			}
+			// When err != nil, the error will be returned at the end
 		} else {
-			log.Warnf("%v token manager specified but failed to ready GCP project information", GoogleTokenExchange)
+			return nil, fmt.Errorf("%v token manager specified but failed to ready GCP project information", GoogleTokenExchange)
 		}
 	}
-	return tm
+	return tm, err
```

Comment:
```suggestion
	return tm, nil
```

Reply:
Because "if p, err := google.CreateTokenManagerPlugin(...)" may return an error, here returns "tm, err", instead of "tm, nil".

Reply:
I have refactored code to return "tm, nil" here.